edb/edgeql/compiler/stmt.py

```diff
@@ -598,6 +608,17 @@ def compile_UpdateQuery(
                 context=expr.subject.context)
 
         mat_stype = schemactx.concretify(subj_type, ctx=ctx)
+
+        if (
+            not (ctx.env.options.bootstrap_mode or
+                 ctx.env.options.schema_reflection_mode)
+            and mat_stype.is_stdlib_type(ctx.env.schema)
+        ):
+            raise errors.QueryError(
+                f'cannot update standard library type '
```

Comment:
Can you explain why? I was hoping we would infer `(BaseObject & schema::Type)` for that. Is that not the case?

Reply:
I think this will just reject that because you `BaseObject` is a stdlib type.

Reply:
> I think this will just reject that because you `BaseObject` is a stdlib type.

Hm, well `std::Object` is also a stdlib type.  Banning mutation of that is a compatibility break and also dubious from the semantic perspective.